### PR TITLE
#2247 Added a Get Single TeamType Endpoint

### LINKED ISSUE #2247: getSingleTeamType endpoint
### Description
We need an endpoint so that users can get a single team type in the database.

### Acceptance Criteria
- Add all these things in the work package files
- This endpoint should be on the route '/teamType/:teamTypeId/single'
- Anyone should be able to get a team type

### Proposed Solution
Look at other single get endpoints for an example

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

Added a getSingleTeamType Endpoint where it retrieves a single teamType from the database.

## Screenshots

image

Exception Handling:
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #2247

### CODE REVIEW

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Great work on this!

Now you just gotta write tests

File: src/backend/tests/unmocked/team-type.test.ts [hunk 1]
```diff
@@ -42,4 +42,17 @@ describe('Team Type Tests', () => {
...
+  describe('Get a single team type', () => {
+    it('Get a single team type works', async () => {
+      const teamType1 = await TeamsService.createTeamType(superman, 'teamType1', 'YouTubeIcon');
+      const result = await TeamsService.getSingleTeamType(teamType1.teamTypeId);
+      expect(result).toStrictEqual(teamType1);
+    });
+
+    it('Get a single team type fails', async () => {
+      const nonExistingTeamTypeId = 'nonExistingId';
+      await expect(async () => TeamsService.getSingleTeamType(nonExistingTeamTypeId)).rejects.toThrow(NotFoundException);
```
Comment (Aaryan1203) [inline src/backend/tests/unmocked/team-type.test.ts:55]: Throw a `new NotFoundException` and make sure it has the exact description of the exception that is being thrown. Look above at other tests that throw exceptions for an example

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Great work, just one small comment

**Review by Aaryan1203 — APPROVED** [review]
Good work
LGTM!

**Review by Aaryan1203 — APPROVED** [review]
great work LGTM

**Review by Peyton-McKee — APPROVED** [review]

**Pull request merged**